src/main/java/madstax/model/util/ResourceIO.java

```diff
@@ -64,12 +65,18 @@ public static ImageIcon getImageIcon(String imageName) {
 
     public static <T> void writeDataToCSV(String fileName, List<T> data, Class<? extends T> type) {
         String csvFileRelativePath = String.format("csv/%s.csv", fileName);
-        ClassLoader classLoader = ResourceIO.class.getClassLoader();
-        URL resourceURL = classLoader.getResource(csvFileRelativePath);
 
         try {
-            assert resourceURL != null;
-            Path filePath = Paths.get(resourceURL.toURI());
+            Path filePath;
+            if (type == Teacher.class) {
```

Comment:
(for my own learning) why is it == and not .equals() in this instance? Just so I know <3

Reply:
I'm comparing class types! In other words, references. So the operator `==` will suffice. We use `.equals()` in Java to compare the content of `Objects`.

https://www.geeksforgeeks.org/difference-equals-method-java/

Reply:
aight 💯